src/core/trulens/experimental/otel_tracing/core/instrument.py

```
@@ -0,0 +1,130 @@
+                    exception = e
+
+                try:
+                    attributes_to_add = {}
+
+                    # Since we're decoratoring a method in a trulens app, the first argument is self,
+                    # which we should ignore.
+                    _self, *rest = args
+
+                    # Set the user provider attributes.
+                    if attributes:
+                        if callable(attributes):
+                            attributes_to_add = attributes(
+                                ret, exception, *rest, **kwargs
+                            )
+                        else:
+                            attributes_to_add = attributes
+
+                    logger.info(f"Attributes to add: {attributes_to_add}")
+
+                    final_attributes = _validate_attributes(attributes_to_add)
+
+                    prefix = "trulens."
+                    if (
+                        SpanAttributes.SPAN_TYPE in final_attributes
+                        and final_attributes[SpanAttributes.SPAN_TYPE]
+                        != SpanAttributes.SpanType.UNKNOWN
+                    ):
+                        prefix += (
+                            final_attributes[SpanAttributes.SPAN_TYPE] + "."
+                        )
+
+                    for key, value in final_attributes.items():
+                        span.set_attribute(prefix + key, value)
+
+                        if (
+                            key != SpanAttributes.SELECTOR_NAME_KEY
+                            and SpanAttributes.SELECTOR_NAME_KEY
+                            in final_attributes
+                        ):
+                            span.set_attribute(
+                                f"trulens.{final_attributes[SpanAttributes.SELECTOR_NAME_KEY]}.{key}",
+                                value,
+                            )
+
+                except Exception as e:
+                    logger.error(f"Error setting attributes: {e}")
```

I think we should throw this error so that a user knows if there's something wrong with their telemetry. Also, we should throw the `exception` we got on L77 if it exists since we shouldn't just swallow all errors.